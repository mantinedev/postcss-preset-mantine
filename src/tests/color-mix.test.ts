import { testTransform } from './utils';

const alphaInput = `
.demo {
  background: alpha(#f00, 0.5);
  border: rem(1px) solid alpha(var(--mantine-color-gray-4), 0.1);
}
`;

const lightenInput = `
.demo {
  background: lighten(#f00, 0.5);
  border: rem(1px) solid lighten(var(--mantine-color-gray-4), 0.1);
}
`;

const darkenInput = `
.demo {
  background: darken(#f00, 0.5);
  border: rem(1px) solid darken(var(--mantine-color-gray-4), 0.1);
}
`;

describe('color-mix', () => {
  it('correctly replaces alpha function', async () => {
    const res = await testTransform(alphaInput);
    expect(res.css).toMatchSnapshot();
  });

  it('correctly replaces lighten function', async () => {
    const res = await testTransform(lightenInput);
    expect(res.css).toMatchSnapshot();
  });

  it('correctly replaces darken function', async () => {
    const res = await testTransform(darkenInput);
    expect(res.css).toMatchSnapshot();
  });
});
