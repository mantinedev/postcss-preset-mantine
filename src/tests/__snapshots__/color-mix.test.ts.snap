// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color-mix correctly replaces alpha function 1`] = `
"
.demo {
  background: color-mix(in srgb, #f00, transparent 50%);
  border: calc(0.0625rem * var(--mantine-scale)) solid color-mix(in srgb, var(--mantine-color-gray-4), transparent 90%);
}
"
`;

exports[`color-mix correctly replaces darken function 1`] = `
"
.demo {
  background: color-mix(in srgb, #f00, black 50%);
  border: calc(0.0625rem * var(--mantine-scale)) solid color-mix(in srgb, var(--mantine-color-gray-4), black 10%);
}
"
`;

exports[`color-mix correctly replaces lighten function 1`] = `
"
.demo {
  background: color-mix(in srgb, #f00, white 50%);
  border: calc(0.0625rem * var(--mantine-scale)) solid color-mix(in srgb, var(--mantine-color-gray-4), white 10%);
}
"
`;
